flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerImplTest.java

```diff
@@ -18,166 +18,187 @@
-				Matchers.anyInt())).thenReturn(sampleFuture);
-
-		ExecutionGraph graph = Mockito.mock(ExecutionGraph.class);
-		Mockito.when(graph.getState()).thenReturn(JobStatus.RUNNING);
-
-		// Same Thread execution context
-		Mockito.when(graph.getFutureExecutor()).thenReturn(new Executor() {
-
-			@Override
-			public void execute(Runnable runnable) {
-				runnable.run();
-			}
-		});
-
-		ExecutionVertex[] taskVertices = new ExecutionVertex[4];
-
-		ExecutionJobVertex jobVertex = Mockito.mock(ExecutionJobVertex.class);
-		Mockito.when(jobVertex.getJobId()).thenReturn(new JobID());
-		Mockito.when(jobVertex.getJobVertexId()).thenReturn(new JobVertexID());
-		Mockito.when(jobVertex.getGraph()).thenReturn(graph);
-		Mockito.when(jobVertex.getTaskVertices()).thenReturn(taskVertices);
-
-		taskVertices[0] = mockExecutionVertex(jobVertex, 0);
-		taskVertices[1] = mockExecutionVertex(jobVertex, 1);
-		taskVertices[2] = mockExecutionVertex(jobVertex, 2);
-		taskVertices[3] = mockExecutionVertex(jobVertex, 3);
-
-		int numSamples = 100;
-		Time delayBetweenSamples = Time.milliseconds(100L);
-
-		BackPressureStatsTrackerImpl tracker = new BackPressureStatsTrackerImpl(
-				sampleCoordinator, 9999, numSamples, Integer.MAX_VALUE, delayBetweenSamples);
-
-		// getOperatorBackPressureStats triggers stack trace sampling
-		Assert.assertFalse(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		Mockito.verify(sampleCoordinator, Mockito.times(1)).triggerStackTraceSample(
-				Matchers.eq(taskVertices),
-				Matchers.eq(numSamples),
-				Matchers.eq(delayBetweenSamples),
-				Matchers.eq(BackPressureStatsTrackerImpl.MAX_STACK_TRACE_DEPTH));
-
-		// Request back pressure stats again. This should not trigger another sample request
-		Assert.assertTrue(!tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		Mockito.verify(sampleCoordinator, Mockito.times(1)).triggerStackTraceSample(
-				Matchers.eq(taskVertices),
-				Matchers.eq(numSamples),
-				Matchers.eq(delayBetweenSamples),
-				Matchers.eq(BackPressureStatsTrackerImpl.MAX_STACK_TRACE_DEPTH));
-
-		Assert.assertTrue(!tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		// Complete the future
-		Map<ExecutionAttemptID, List<StackTraceElement[]>> traces = new HashMap<>();
-		for (ExecutionVertex vertex : taskVertices) {
-			List<StackTraceElement[]> taskTraces = new ArrayList<>();
+	public void testGetOperatorBackPressureStats() throws Exception {
+		doInitialRequestAndVerifyTheResult(createBackPressureTracker(60000, 60000, backPressureStats));
+	}
 
-			for (int i = 0; i < taskVertices.length; i++) {
-				// Traces until sub task index are back pressured
-				taskTraces.add(createStackTrace(i <= vertex.getParallelSubtaskIndex()));
-			}
+	@Test
+	public void testCachedStatsNotUpdatedWithinRefreshInterval() throws Exception {
+		final double backPressureRatio2 = 0.2;
+		final BackPressureStats backPressureStats2 = createBackPressureStats(1, 60000, backPressureRatio2);
```

Comment:
It is better to define these variables to use in the methods for better understanding the meanings.

Because there are two many similar numbers `1`, `60000` used in different methods `createBackPressureStats`, `createBackPressureTracker`, and I was almost confused every time to understand what 's the meaning for each number and had to go through the specific below methods to trace them.